File: src/main/java/com/github/kayjamlang/core/expressions/UseExpression.java
package com.github.kayjamlang.core.expressions;

import com.github.kayjamlang.core.opcodes.AccessType;

import java.util.List;

public class UseExpression extends Expression {
    public List<String> required;
    public final String from;

    public UseExpression(List<String> required, String from, int line){
        super(AccessType.NONE, line);
        this.required = required;
        this.from = from;

        expression = null;
    }

    /**
     * @deprecated please don't use this
     */
    @Deprecated
    public final Expression expression;

    /**
     * @deprecated please don't use this
     */
    @Deprecated
    public UseExpression(Expression expression, int line) {
        super(AccessType.NONE, line);
        this.expression = expression;
        this.from = null;
    }

    /**
     * @deprecated please don't use this
     */
    @Deprecated
    public interface UseInterface{
        Expression getExpression(String path) throws Exception;
    }
}

File: src/main/scala/com/github/kayjamlang/core/expressions/UseExpression.scala
package com.github.kayjamlang.core.expressions

import com.github.kayjamlang.core.ArrayList
import com.github.kayjamlang.core.opcodes.AccessType

import scala.collection.mutable

class UseExpression(line: Int, var required: ArrayList[String], val from: String) extends Expression(AccessType NONE, line) {
  def this(required: ArrayList[String], from: String, line: Int) {
    this(line, required, from)
  }
}
